robot_name/robot_name.rb

```diff
@@ -2,31 +2,48 @@ class NameCollisionError < RuntimeError; end
 
 class Robot
   attr_accessor :name
-
-  @@registry
+  @@registry = []
 
   def initialize(args = {})
-    @@registry ||= []
-    @name_generator = args[:name_generator]
-
-    if @name_generator
-      @name = @name_generator.call
-    else
-      generate_char = -> { ('A'..'Z').to_a.sample }
-      generate_num = -> { rand(10) }
+    @name = args[:name] || generate_name
```

Comment:
Big issue here: you changed the method signature. Previously, the Robot class accepted a `name_generator` but you changed it to only accept a new `name` instead. When refactoring, you should try to limit the functional changes as much as possible.